### PR TITLE
updated readme

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
## Describe your changes

Updates the readme to be more comprehensive. Note that the base branch is `use-vite` instead of `dev`, this will be updated once `use-vite` is merged, or if this request is merged first. 

## Issue ticket number and Jira link

[REMS-698](https://jira.mitre.org/browse/REMS-698)
## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] Ensure the target / base branch for any feature PR is set to `dev` not main (the only exception to this is releases from `dev` and hotfix branches)

## Checklist for conducting a review
- [ ] Review the code changes and make sure they all make sense and are necessary. 
- [ ] Pull the PR branch locally and test by running through workflow and making sure everything works as it is supposed to.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,4 +47,48 @@ It correctly bundles React in production mode and optimizes the build for the be
...
+## Environment Variables
+
+The .env file contains the default URI paths, which can be overwritten from the start command as follows:
+a) `REACT_APP_LAUNCH_URL=http://example.com PORT=6000 npm start` or b) by specifying the environment variables and desired values in a `.env.local`.
+
+Following are a list of modifiable paths:
+
+| URI Name                          | Default Value                                                                         | Description                                     |
+| --------------------------------- | ------------------------------------------------------------------------------------- | ----------------------------------------------- |
+| GENERATE_SOURCEMAP                | `false`                                                                               | Set to 'true' to generate a sourcemap.  A sourcemap allows the browser to reference untranspiled code.  This is useful for debugging and developing, but should not be used in production.                                                  |
```
[the remark this answers is not in the document]
Comment (jtquach1) [reply README.md:83]: but also, I am not really sure how GENERATE_SOURCEMAP works

**Review by jtquach1 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -59,4 +47,48 @@ It correctly bundles React in production mode and optimizes the build for the be
 The build is minified and the filenames include the hashes.\
 Your app is ready to be deployed!
 
-See the section about [deployment](https://create-react-app.dev/docs/deployment/) for more information.
+See the section about [building and deploying](https://vitejs.dev/guide/build) for more information.
+
+
+## Usage
+
+The REMS SMART on FHIR app interacts with the [REMS Admin](https://github.com/mcode/rems-admin), an [EHR](https://github.com/mcode/test-ehr), and the [Pharmacy Information Management System](https://github.com/mcode/pims).  These apps are provided as part of the REMS ecosystem, but any individual part may be swapped out for something custom.  The EHR in particular can be easily switched by launching the SMART app from a different EHR.  Launching the app can be easily done through the [Request Generator](https://github.com/mcode/request-generator), a front end app for EHRs that sends and recieves CDS Hooks.  The SMART app must be launched through an EHR or through the Request Generator.
```
Comment (plarocque4) [inline README.md:55]: Does the SMART on FHIR app still interact with PIMS? Checking the pharmacy status involves a check to the EHR FHIR server.

File: README.md [hunk 3]
```diff
@@ -59,4 +47,48 @@ It correctly bundles React in production mode and optimizes the build for the be
...
+
+Typically, a CDS Hook will be sent from the EHR to the REMS Admin, which will respond with cards that contain information about next steps. These cards may contain a link to a SMART app. Clicking on these links will launch the SMART app automatically.
```
Comment (plarocque4) [inline README.md:57]: Should we mention something about checking ETASU and pharmacy status?

File: README.md [hunk 4]
```diff
@@ -59,4 +47,48 @@ It correctly bundles React in production mode and optimizes the build for the be
...
+| REACT_APP_PHARMACY_STATUS_ENABLED | `true`                                                                                | When set to 'true', shows the pharmacy status on the main page. This allows the user to track progress towards dispensing the medication from the pharmacy.                                                 |
+| REACT_APP_REMS_ADMIN_SERVER_BASE  | `http://localhost:8090`                                                               | The base url of the REMS admin server, which handles the ETASU and questionnaires. Should be changed to match the base url of the REMS admin you wish to submit information to.                   |
+| REACT_APP_SEND_FHIR_AUTH_ENABLED  | `false`                                                                               | When set to 'true', the app will send the access token for the EHR to the REMS admin as part of the CDS Hook.  Should be changed to false only if this functionality is required for the REMS admin to work, and is a trusted party.                                                |
+
+_Note that .env values can only be accessed by the React app starting with `REACT_APP_`\_
```
[the remark this answers is not in the document]
Comment (KeeyanGhoreshi) [reply README.md:94]: I didn't write this bit, just moved it, it's not clear to me what form of font adornment it needs

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (KeeyanGhoreshi) [reply README.md:83]: as far as I can tell vite generates the sourcemaps in dev either way, so the build option in the config file doesn't change anything.  We might be able to just remove this env variable altogether.

File: README.md (hunk 2 above)
Comment (jtquach1) [reply README.md]: It does not, the env var was deleted; the src/views/Patient/MedReqDropDown/MedReqDropDown.tsx and src/views/Patient/MedReqDropDown/pharmacyStatus/PharmacyStatus.tsx use the response from test-ehr's MedicationRequest endpoint

File: README.md [hunk 5]
```diff
@@ -59,4 +47,48 @@ It correctly bundles React in production mode and optimizes the build for the be
 The build is minified and the filenames include the hashes.\
 Your app is ready to be deployed!
 
-See the section about [deployment](https://create-react-app.dev/docs/deployment/) for more information.
+See the section about [building and deploying](https://vitejs.dev/guide/build) for more information.
+
+
+## Usage
+
+The REMS SMART on FHIR app interacts with the [REMS Admin](https://github.com/mcode/rems-admin), and an [EHR](https://github.com/mcode/test-ehr).  These apps are provided as part of the REMS ecosystem, but any individual part may be swapped out for something custom.  The EHR in particular can be easily switched by launching the SMART app from a different EHR.  Launching the app can be easily done through the [Request Generator](https://github.com/mcode/request-generator), a front end app for EHRs that sends and recieves CDS Hooks.  The SMART app must be launched through an EHR or through the Request Generator.
```
Comment (jtquach1) [inline README.md:55]:
```suggestion
The REMS SMART on FHIR app interacts with the [REMS Admin](https://github.com/mcode/rems-admin), and an [EHR](https://github.com/mcode/test-ehr).  These apps are provided as part of the REMS ecosystem, but any individual part may be swapped out for something custom.  The EHR in particular can be easily switched by launching the SMART app from a different EHR.  Launching the app can be easily done through the [Request Generator](https://github.com/mcode/request-generator), a front end app for EHRs that sends and receives CDS Hooks.  The SMART app must be launched through an EHR or through the Request Generator.
```

**Review by jtquach1 — APPROVED** [review]

**Pull request merged**